eng/templates/stages/release.yml

```
@@ -71,243 +106,242 @@ stages:
-        title=$(echo "$template" | grep "^Title:" | cut -d " " -f2-)
-        # Get the inverse of the above selection
-        body=$(echo "$template" | grep -v "^Title:")
-
-        echo "TargetRepo: $(TargetRepo)"
-        echo "ForkRepo: $(ForkRepo)"
-        echo "SdkVersion: $(sdkVersion)"
-        echo "title: ${title}"
-        echo "body: ${body}"
-
-        extraArgs=()
-        if [[ "$(releaseChannel)" == '6.0' || "$(releaseChannel)" == '7.0' ]]; then
-          extraArgs+=("--globalJson" "src/SourceBuild/tarball/content/global.json")
-          extraArgs+=("--versionProps" "eng/Versions.props")
-        fi
-
-        if [ ${{ parameters.isDryRun }} = True ]; then
-          echo "Doing a dry run, not submitting PR. Would have called:"
-          echo "./submit-source-build-release-pr.sh"
-          echo "  --setupGitAuth"
-          echo "  --targetRepo $(TargetRepo)"
-          echo "  --forkRepo $(ForkRepo)"
-          echo "  --sdkVersion $(sdkVersion)"
-          echo "  --title ${title}"
-          echo "  --body ${body}"
-          echo "  --targetBranch ${{ parameters.releaseBranchName }}"
-          echo "  ${extraArgs[@]}"
-        else
-          echo "Submitting PR"
-          ./submit-source-build-release-pr.sh \
-            --setupGitAuth \
-            --targetRepo $(TargetRepo) \
-            --forkRepo $(ForkRepo) \
-            --sdkVersion $(sdkVersion) \
-            --title "${title}" \
-            --body "${body}" \
-            --targetBranch "${{ parameters.releaseBranchName }}" \
-            "${extraArgs[@]}"
-        fi
-      displayName: Submit Release PR
-      workingDirectory: $(Build.SourcesDirectory)/eng
-      env:
-        GH_TOKEN: $(BotAccount-dotnet-sb-bot-pat)
+    - checkout: self
+
+    - ${{ if and(ne(parameters.createGitHubRelease, 'skip'), ne(parameters.dotnetMajorVersion, '6.0'), ne(parameters.dotnetMajorVersion, '7.0')) }}:
+      - checkout: dotnet-dotnet
+        displayName: Checkout dotnet-dotnet
+        fetchDepth: 0
+        fetchTags: true
+
+      - script: |
+          set -euo pipefail
+          tarball_path="$(Build.ArtifactStagingDirectory)/dotnet-sdk-source-$(release).tar.gz"
+          echo "Creating source tarball from dotnet-dotnet / $(dotnetDotnetCommit)"
+
+          git checkout "$(dotnetDotnetCommit)"
+          ./eng/pack-sources.sh --output "$tarball_path"
+
+          echo "##vso[task.setvariable variable=SourceTarballPath]$tarball_path"
+
+        displayName: Prepare source tarball
+        workingDirectory: $(Build.SourcesDirectory)/dotnet-dotnet
+
+      - script: |
+          set -euo pipefail
+
+          prerelease=''
+          if [ ${{ parameters.isPreviewRelease }} = True ]; then
+            prerelease='--prerelease'
+          fi
+
+          draft=''
+          if [ ${{ parameters.createGitHubRelease }} = draft ]; then
+            draft='--draft'
+          fi
+
+          gh release create "$(releaseTag)" "$(SourceTarballPath)#Packed sources" \
```

IMO we should be using the standard GH name here and give an indication it is a tarball.

```suggestion
          gh release create "$(releaseTag)" "$(SourceTarballPath)#Source code (tar.gz)" \
```